src/cargo/ops/cargo_doc.rs

```diff
@@ -61,38 +67,47 @@ pub fn doc(manifest_path: &Path,
 }
 
 #[cfg(not(any(target_os = "windows", target_os = "macos")))]
-fn open_docs(path: &Path) {
+fn open_docs(path: &Path) -> Result<(), ()> {
+    // trying $BROWSER
+    match env::var("BROWSER").map(|name| Command::new(name).arg(path).status()) {
```

Comment:
Out of curiosity, is this a standard method of specifying a program to open something with?